pkg/cmd/infra/router/template_test.go

```diff
@@ -0,0 +1,41 @@
+package router
+
+import (
+	"github.com/google/go-cmp/cmp"
+	templateplugin "github.com/openshift/router/pkg/router/template"
+	"testing"
+)
+
+func TestPercentDecodingForHeaders(t *testing.T) {
+	testCases := []struct {
+		description   string
+		inputValue    string
+		expectedValue []templateplugin.HTTPHeader
+		expectError   bool
+	}{
+		{
+			description: "should percent decode the header values",
+			inputValue:  "X-Frame-Options:DENY:Set,X-XSS-Protection:1%3Bmode%3Dblock:Set,x-forwarded-client-cert:%25%7B%2BQ%7D%5Bssl_c_der%2Cbase64%5D:Set,X-Frame-Options:Delete,X-XSS-Protection:Delete",
+			expectedValue: []templateplugin.HTTPHeader{
+
+				{Name: "X-Frame-Options", Value: "'DENY'", Action: "Set"},
+				{Name: "X-XSS-Protection", Value: "'1;mode=block'", Action: "Set"},
+				{Name: "x-forwarded-client-cert", Value: "'%{+Q}[ssl_c_der,base64]'", Action: "Set"},
+				{Name: "X-Frame-Options", Action: "Delete"},
+				{Name: "X-XSS-Protection", Action: "Delete"},
+			},
+			expectError: false,
+		},
+	}
```

Comment:
I agree with Candace's comment about this test.  You have a list of test cases, but you only have one test case in that list.  The test case that you do have verifies that the function parses valid input.  You need to add some test cases with invalid input, so that your test verifies that `parseHeadersToBeSetOrDeleted` behaves correctly when given invalid input.  There are several places in `parseHeadersToBeSetOrDeleted` that return errors, so you should have some test cases where `expectError` is true.

Reply:
ack added tests to check error messages.